### ISSUE TITLE
Go further with architecture.

### ISSUE THREAD

**Issue by proofit404** [proofit404/cruftbot#251]
Initially, I thought that bet on `pydantic` was a good idea.

Now I would favor to have `attrs` as entities and services.

`stories` would use plain `State` contracts.

`stories_marshmallow` would provide validation functions.

I still don't know what to use for validators.

Eventually, `generics` `@defended` decorator would be implemented.

`primitives` should be used to test services.

`flake8-no-typing` should be enabled.

`import linter` should be enabled (`entities` -> `services` -> `implemented` -> `infrastructure`).

#### Layout
* cruftbot
    * entities
    * usecases
        * validation (marshmallow schema to instantiate entities)
        * states
        * services
        * implemented (link state to services, reused in own test cases)
    * infrastructure
        * repositories
        * django project
        * implemented (extends definitions from usecases one)
* tests
    * entities (unit tests for entity classes)
    * usecases (dependencies + primitives tests)
    * infrastructure
        * repositories (contract tests generated from primitives)
        * django view tests

**Comment by proofit404** [comment]
After taking a deep breath I think we could use pydantic to some degree:

1. Entities
```python
from generics import private
from pydantic.dataclasses import dataclass

@private
@dataclass
class User:
    name: str

    def greet(self):
        ...
```
2. Use cases
```python
from typing import Callable
from pydantic.dataclasses import dataclass
from stories import Story, I, State
from stories_pydantic import Argument, Variable

@dataclass
class Subscribe(Story):
    I.find_category
    I.find_profile
    I.check_balance
    I.persist_subscription

    def find_category(self, state):
        state.category = self.load_category(state.category_id)

    def find_profile(self, state):
        state.profile = self.load_profile(state.profile_id)

    def check_balance(self, state):
        if not state.category.affordable_for(state.profile):
            raise Exception

    def persist_subscription(self, state):
        state.subscription = self.create_subscription(category=state.category_id, profile=state.profile_id)

    load_category: Callable
    load_profile: Callable
    create_subscription: Callable # check DI interfaces until `generics.protected` is not ready

class SubscribeState(State):
    category_id = Argument(int)
    profile_id = Argument(int) # infer validation from `int` type
    profile = Variable(User) # convert dictionary to entity instance
```

**Comment by 0pdd** [comment]
@proofit404 2 puzzles [#268](https://github.com/proofit404/cruftbot/issues/268), [#269](https://github.com/proofit404/cruftbot/issues/269) are still not solved.

**Comment by 0pdd** [comment]
@proofit404 2 puzzles [#268](https://github.com/proofit404/cruftbot/issues/268), [#299](https://github.com/proofit404/cruftbot/issues/299) are still not solved; solved: [#269](https://github.com/proofit404/cruftbot/issues/269).

**Comment by 0pdd** [comment]
@proofit404 the puzzle [#299](https://github.com/proofit404/cruftbot/issues/299) is still not solved; solved: [#268](https://github.com/proofit404/cruftbot/issues/268), [#269](https://github.com/proofit404/cruftbot/issues/269).

**Comment by 0pdd** [comment]
@proofit404 all 3 puzzles are solved here: [#268](https://github.com/proofit404/cruftbot/issues/268), [#269](https://github.com/proofit404/cruftbot/issues/269), [#299](https://github.com/proofit404/cruftbot/issues/299).
